### ISSUE TITLE
Cannot export submissions

### ISSUE THREAD

**Issue by JulieVelghe** [verbb/formie#966]
**Description**
When I select my submissions to export to csv, I get a server error
![formieerror](https://user-images.githubusercontent.com/46524074/176916383-ab16e0c1-063a-4737-9f52-047a28af911e.png)

**Steps to reproduce**
1.
2.

### Form settings
Please pick from the options below.

- Multi-page form: No
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: No

**Additional info**
- Plugin version: 2.0.0-beta.14
- Craft version:4.1.0.2
- Multi-site: No

**Additional context**

**Comment by engram-design** [comment]
Should be fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-4 as 2.0.0-beta.16",
  "...": "..."
}
```

Then run `composer update`.

**Comment by JulieVelghe** [comment]
Hi, i've tried this new version, but the bug is still there

**Comment by engram-design** [comment]
If you click on that error link in the console, it should take you to the Network tab. Can you look at the "Preview" tab to see what the error being reported is?

![image](https://user-images.githubusercontent.com/1221575/177100607-9bcbe63c-2fdd-4819-b24d-ff199f6e132d.png)

**Comment by JulieVelghe** [comment]
it's a 500 i believe:
![image](https://user-images.githubusercontent.com/46524074/177101181-c24cb85d-d9d9-48ce-a9e4-b8e41a67f887.png)

**Comment by engram-design** [comment]
Are you able to turn on `devMode`? Or if you can't, have a look in your logs for the full stack trace of the error at `storage/logs/*.log`

**Comment by JulieVelghe** [comment]
Yes, my local environment is in devMode. I get the following:
An Error occurred while handling another error:
TypeError: array_keys(): Argument #1 ($array) must be of type array, string given in C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\craftcms\cms\src\web\CsvResponseFormatter.php:88
Stack trace:
#0 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\craftcms\cms\src\web\CsvResponseFormatter.php(88): array_keys('Exception')
#1 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\yiisoft\yii2\web\Response.php(1095): craft\web\CsvResponseFormatter->format(Object(craft\web\Response))
#2 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\craftcms\cms\src\web\Response.php(286): yii\web\Response->prepare()
#3 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\yiisoft\yii2\web\Response.php(339): craft\web\Response->prepare()
#4 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\yiisoft\yii2\web\ErrorHandler.php(136): yii\web\Response->send()
#5 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\craftcms\cms\src\web\ErrorHandler.php(194): yii\web\ErrorHandler->renderException(Object(ArgumentCountError))
#6 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\yiisoft\yii2\base\ErrorHandler.php(135): craft\web\ErrorHandler->renderException(Object(ArgumentCountError))
#7 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\craftcms\cms\src\web\ErrorHandler.php(64): yii\base\ErrorHandler->handleException(Object(ArgumentCountError))
#8 [internal function]: craft\web\ErrorHandler->handleException(Object(ArgumentCountError))
#9 {main}
Previous exception:
ArgumentCountError: array_merge() does not accept unknown named parameters in C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\verbb\formie\src\elements\Submission.php:894
Stack trace:
#0 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\verbb\formie\src\elements\Submission.php(894): array_merge(naam: Array, huidigeOrganisatie: Array, voorstelNieuweNaam: Array, denkpiste: Array)
#1 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\verbb\formie\src\elements\exporters\SubmissionExport.php(67): verbb\formie\elements\Submission->getValuesForExport()
#2 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\craftcms\cms\src\controllers\ElementIndexesController.php(324): verbb\formie\elements\exporters\SubmissionExport->export(Object(verbb\formie\elements\db\SubmissionQuery))
#3 [internal function]: craft\controllers\ElementIndexesController->actionExport()
#4 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\yiisoft\yii2\base\InlineAction.php(57): call_user_func_array(Array, Array)
#5 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\yiisoft\yii2\base\Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#6 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\yiisoft\yii2\base\Module.php(552): yii\base\Controller->runAction('export', Array)
#7 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\craftcms\cms\src\web\Application.php(301): yii\base\Module->runAction('element-indexes...', Array)
#8 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\craftcms\cms\src\web\Application.php(625): craft\web\Application->runAction('element-indexes...', Array)
#9 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\craftcms\cms\src\web\Application.php(280): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#10 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\yiisoft\yii2\base\Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#11 C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\www\index.php(12): yii\base\Application->run()
#12 {main}

How I try to get my export: I select all the submissions, and then choose default export type and csv format

**Comment by engram-design** [comment]
Can you please run the `composer update` command again? It looks like the code is still referring to the code before my fix. For example this line in the error:

```
ArgumentCountError: array_merge() does not accept unknown named parameters in C:\Users\julie\Documents\CabinetC\Repos\gentsewoonmaatschappij\vendor\verbb\formie\src\elements\Submission.php:894
```

Doesn't exist as you can see below:
https://github.com/verbb/formie/blob/9324da68c423ac88e151f5cd18169e7a94c8ca7b/src/elements/Submission.php#L894

**Comment by JulieVelghe** [comment]
I did a composer update, but still the same error. Would it help if I removed the vendor map and then did composer install again? Would I lose any of the submissions data?

**Comment by engram-design** [comment]
Yep, maybe remove your `composer.lock` file and delete the `vendor` directory, and make sure you use `composer update` (not just `composer install`), as it sounds like it's not fetching the latest updates.

**Comment by JulieVelghe** [comment]
yes, that did it :)

**Comment by JulieVelghe** [comment]
when the fix gets released, will I have to change back the requirement?

**Comment by engram-design** [comment]
Yep, you'll want to run `composer require verbb/formie`. We tag released most weekends.

**Comment by JeanLucEsser** [comment]
Hey @engram-design, not sure if this is related but I have an error on export as well, per the php logs:

```
[04-Jul-2022 17:35:26 Europe/Paris] An Error occurred while handling another error:
TypeError: array_keys(): Argument #1 ($array) must be of type array, string given in /var/www/html/vendor/craftcms/cms/src/web/CsvResponseFormatter.php:88
Stack trace:
#0 /var/www/html/vendor/craftcms/cms/src/web/CsvResponseFormatter.php(88): array_keys('Exception')
#1 /var/www/html/vendor/yiisoft/yii2/web/Response.php(1095): craft\web\CsvResponseFormatter-format(Object(craft\web\Response))
#2 /var/www/html/vendor/craftcms/cms/src/web/Response.php(286): yii\web\Response->prepare()
#3 /var/www/html/vendor/yiisoft/yii2/web/Response.php(339): craft\web\Response->prepare()
#4 /var/www/html/vendor/yiisoft/yii2/web/ErrorHandler.php(136): yii\web\Response->send()
#5 /var/www/html/vendor/craftcms/cms/src/web/ErrorHandler.php(194): yii\web\ErrorHandler-renderException(Object(TypeError))
#6 /var/www/html/vendor/yiisoft/yii2/base/ErrorHandler.php(135): craft\web\ErrorHandler-renderException(Object(TypeError))
#7 /var/www/html/vendor/craftcms/cms/src/web/ErrorHandler.php(64): yii\base\ErrorHandler-handleException(Object(TypeError))
#8 [internal function]: craft\web\ErrorHandler->handleException(Object(TypeError))
#9 {main}
Previous exception:
TypeError: verbb\formie\fields\formfields\Entries::_all(): Argument #1 ($query) must be of type craft\elements\db\ElementQueryInterface, Illuminate\Support\Collection given, called in /var/www/html/vendor/verbb/formie/src/base/RelationFieldTrait.php on line 289 and defined in /var/www/html/vendor/verbb/formie/src/base/RelationFieldTrait.php:372
Stack trace:
#0 /var/www/html/vendor/verbb/formie/src/base/RelationFieldTrait.php(289): verbb\formie\fields\formfields\Entries-_all(Object(Illuminate\Support\Collection), Object(verbb\formie\elements\Submission))
#1 /var/www/html/vendor/verbb/formie/src/base/FormFieldTrait.php(1132): verbb\formie\fields\formfields\Entries-defineValueAsString(Object(Illuminate\Support\Collection), Object(verbb\formie\elements\Submission))
#2 /var/www/html/vendor/verbb/formie/src/base/FormFieldTrait.php(215): verbb\formie\fields\formfields\Entries-defineValueForExport(Object(Illuminate\Support\Collection), Object(verbb\formie\elements\Submission))
#3 /var/www/html/vendor/verbb/formie/src/elements/Submission.php(886): verbb\formie\fields\formfields\Entries-getValueForExport(Object(Illuminate\Support\Collec
tion), Object(verbb\formie\elements\Submission))
#4 /var/www/html/vendor/verbb/formie/src/elements/exporters/SubmissionExport.php(67): verbb\formie\elements\Submission->getValuesForExport()
#5 /var/www/html/vendor/craftcms/cms/src/controllers/ElementIndexesController.php(324): verbb\formie\elements\exporters\SubmissionExport->export(Object(verbb\formie\elements\db\SubmissionQuery))
#6 [internal function]: craft\controllers\ElementIndexesController->actionExport()
#7 /var/www/html/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#8 /var/www/html/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#9 /var/www/html/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('export', Array)
#10 /var/www/html/vendor/craftcms/cms/src/web/Application.php(301): yii\base\Module->runAction('element-indexes...', Array)
#11 /var/www/html/vendor/craftcms/cms/src/web/Application.php(625): craft\web\Application->runAction('element-indexes...', Array)
#12 /var/www/html/vendor/craftcms/cms/src/web/Application.php(280): craft\web\Application-_processActionRequest(Object(craft\web\Request))
#13 /var/www/html/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application-handleRequest(Object(craft\web\Request))
#14 /var/www/html/web/index.php(12): yii\base\Application->run()
#15 {main}
```

Running dev-craft-4.
Do you want me to open a new issue?

**Comment by engram-design** [comment]
@JeanLucEsser Different issue related to exporting element fields, but should be fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-4 as 2.0.0-beta.16",
  "...": "..."
}
```

Then run `composer update`.

**Comment by JeanLucEsser** [comment]
@engram-design Almost there... this one is fixed, on to the next:

```
[06-Jul-2022 16:06:08 Europe/Paris] An Error occurred while handling another error:
TypeError: array_keys(): Argument #1 ($array) must be of type array, string given in /var/www/html/vendor/craftcms/cms/src/web/CsvResponseFormatter.php:88
Stack trace:
#0 /var/www/html/vendor/craftcms/cms/src/web/CsvResponseFormatter.php(88): array_keys('PHP Warning')
#1 /var/www/html/vendor/yiisoft/yii2/web/Response.php(1095): craft\web\CsvResponseFormatter->format(Object(craft\web\Response))
#2 /var/www/html/vendor/craftcms/cms/src/web/Response.php(286): yii\web\Response->prepare()
#3 /var/www/html/vendor/yiisoft/yii2/web/Response.php(339): craft\web\Response->prepare()
#4 /var/www/html/vendor/yiisoft/yii2/web/ErrorHandler.php(136): yii\web\Response->send()
#5 /var/www/html/vendor/craftcms/cms/src/web/ErrorHandler.php(194): yii\web\ErrorHandler->renderException(Object(yii\base\ErrorException))
#6 /var/www/html/vendor/yiisoft/yii2/base/ErrorHandler.php(135): craft\web\ErrorHandler->renderException(Object(yii\base\ErrorException))
#7 /var/www/html/vendor/craftcms/cms/src/web/ErrorHandler.php(64): yii\base\ErrorHandler->handleException(Object(yii\base\ErrorException))
#8 [internal function]: craft\web\ErrorHandler->handleException(Object(yii\base\ErrorException))
#9 {main}
Previous exception:
yii\base\ErrorException: Attempt to read property "title" on int in /var/www/html/vendor/verbb/formie/src/base/RelationFieldTrait.php:424
Stack trace:
#0 /var/www/html/vendor/craftcms/cms/src/web/ErrorHandler.php(77): yii\base\ErrorHandler->handleError(2, 'Attempt to read...', '/var/www/html/v...', 424)
#1 /var/www/html/vendor/verbb/formie/src/base/RelationFieldTrait.php(424): craft\web\ErrorHandler->handleError(2, 'Attempt to read...', '/var/www/html/v...', 424)
#2 /var/www/html/vendor/verbb/formie/src/base/RelationFieldTrait.php(317): verbb\formie\fields\formfields\Entries->_getElementLabel(26)
#3 [internal function]: verbb\formie\fields\formfields\Entries->verbb\formie\base\{closure}(26)
#4 /var/www/html/vendor/verbb/formie/src/base/RelationFieldTrait.php(316): array_map(Object(Closure), Array)
#5 /var/www/html/vendor/verbb/formie/src/base/FormFieldTrait.php(1301): verbb\formie\fields\formfields\Entries->defineValueAsString(Array, Object(verbb\formie\elements\Submission))
#6 /var/www/html/vendor/verbb/formie/src/base/FormFieldTrait.php(219): verbb\formie\fields\formfields\Entries->defineValueForExport(Object(Illuminate\Support\Collection), Object(verbb\formie\elements\Submission))
#7 /var/www/html/vendor/verbb/formie/src/elements/Submission.php(886): verbb\formie\fields\formfields\Entries->getValueForExport(Object(Illuminate\Support\Collection), Object(verbb\formie\elements\Submission))
#8 /var/www/html/vendor/verbb/formie/src/elements/exporters/SubmissionExport.php(67): verbb\formie\elements\Submission->getValuesForExport()
#9 /var/www/html/vendor/craftcms/cms/src/controllers/ElementIndexesController.php(324): verbb\formie\elements\exporters\SubmissionExport->export(Object(verbb\formie\elements\db\SubmissionQuery))
#10 [internal function]: craft\controllers\ElementIndexesController->actionExport()
#11 /var/www/html/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#12 /var/www/html/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#13 /var/www/html/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('export', Array)
#14 /var/www/html/vendor/craftcms/cms/src/web/Application.php(301): yii\base\Module->runAction('element-indexes...', Array)
#15 /var/www/html/vendor/craftcms/cms/src/web/Application.php(625): craft\web\Application->runAction('element-indexes...', Array)
#16 /var/www/html/vendor/craftcms/cms/src/web/Application.php(280): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#17 /var/www/html/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#18 /var/www/html/web/index.php(12): yii\base\Application->run()
#19 {main}
```

Sorry about that ;)

**Comment by engram-design** [comment]
@JeanLucEsser Can you send through a support request (Formie > Settings > Get Support) so I can see the fields and their setup for your form? Element fields seem to export okay for me.

**Comment by JeanLucEsser** [comment]
Done. Thanx.

Sent the support request from production which does not have the latest dev-4 which I tested on my local environment. Just so you know I did test with the latest. But you should be ok to replicate.

**Comment by engram-design** [comment]
@JeanLucEsser I can't replicate it with your exact form setup, and multiple submission. This seems to be tied to your Entries field "Tessiture" having the value of an integer `26` instead of an element object.

Are you using any template overrides or custom templates for your fields?

**Comment by JeanLucEsser** [comment]
@engram-design No templates overrides or custom templates.

I'm going to try to narrow the problem by re creating that same form field after field and test every step of the way, starting with that "Tessiture" field, but I'm not seeing anything wrong with it as it stands (a test form with only that entry field works fine). Even a single submission export throws the error so it's not like if a specific submission was causing the issue.

**Comment by engram-design** [comment]
Hmmm, there must be something special on your install, as I've tested a whole bunch of different submissions and field combinations on my end. The error seems to point to an issue with an entries field.

**Comment by JeanLucEsser** [comment]
I'm about to recreate the form from scratch (fresh install).
Will report back, here's hoping that something stuck on my end from the beta cycle.

**Comment by JeanLucEsser** [comment]
@engram-design 

FOUND IT !!!!!

Ok, so here it is:

You cannot have a Formie `Entries` field which `Handle` is the same as a Craft Entry field.

I had a Craft Entry Field `tessiture` and my Formie Entries Field Handle was `tessiture`.

My question : is it something you can avoid? Reason is it's quite difficult to keep tabs on the naming of all our fields, and maybe one day someone will add a Craft Field which handle is used in a form somewhere, breaking it in the process. So either we create a routine to having a naming convention for Formie entries fields like appending _formie to those (should document that btw), or you can get around it somehow.

Anyway, FIXED!

**Comment by engram-design** [comment]
Ah, that's interesting! So yes this can be fixed, but this also poses another potential issue. If you're exporting from the "All Forms" page, you _might_ get issues if you have the exact same-named field across multiple forms. This is because of how the querying works when looking up against content. As each form maintains its own content table for submissions, querying across multiple forms and content tables can be tricky.

I'll keep an eye on if this causes problems, but this issue should be fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-4 as 2.0.0",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [2.0.1](https://github.com/verbb/formie/releases/tag/2.0.1)
